### PR TITLE
Support for new cluster app Helm values structure

### PR DESCRIPTION
**Pull request by nprokopic**
### What this PR does

Towards https://github.com/giantswarm/roadmap/issues/2954

### Checklist

- [x] Update changelog in CHANGELOG.md.

### Trigger e2e tests

/run cluster-test-suites

### CODE REVIEW

File: internal/common/clustervalues.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package common
+
+import (
+	"github.com/giantswarm/clustertest/pkg/application"
+)
+
+// ClusterValues holds common values for cluster-<provider> charts. These are
+// the provider independent values and are present for all the charts
+type ClusterValues struct {
```
Comment (nprokopic) [inline internal/common/clustervalues.go:9]: We need to temporarily support both old and new cluster app Helm values schema, so here we're adding a struct for the new schema.

Team Turtles will provide a new go module with structs for cluster app Helm values, as those are used in other places as well, not just in `clustertest` and `cluster-test-suites`, so the new go module will be auto-generated and kept up-to-date with the schema changes.

Until we get the new module (upcoming weeks), here we do a temporary solution for the sake of unblocking cluster-aws work.

File: internal/upgrade/upgrade.go [hunk 2]
```diff
@@ -36,14 +36,22 @@ func Run() {
...
 
 		It("has all the worker nodes running", func() {
-			values := &application.ClusterValues{}
-			err := state.GetFramework().MC().GetHelmValues(cluster.Name, cluster.GetNamespace(), values)
+			var valuesWithOldSchema *application.ClusterValues
+			valuesWithNewSchema := &common.ClusterValues{}
+			err := state.GetFramework().MC().GetHelmValues(state.GetCluster().Name, state.GetCluster().GetNamespace(), valuesWithNewSchema)
 			Expect(err).NotTo(HaveOccurred())
 
+			if common.ClusterIsUsingNewHelmValuesSchema(valuesWithNewSchema) {
```
Comment (nprokopic) [inline internal/upgrade/upgrade.go:44]: If cluster is using the new schema, convert new Helm values to old Helm values, so we don't have to change the rest of the code that is still needed for other providers other than CAPA.

Once all providers have been restructured, here we will revert the changes and use just the new struct for the new cluster app Helm values schema.

File: providers/capa/standard/test_data/cluster_values.yaml [hunk 3]
```diff
@@ -1,23 +1,24 @@
-metadata:
-  name: "{{ .ClusterName }}"
-  description: "E2E Test cluster"
-  organization: "{{ .Organization }}"
+global:
```
Comment (nprokopic) [inline providers/capa/standard/test_data/cluster_values.yaml:1]: For now fixing the standard test. Once there is a new cluster-aws release, we'll fix also the upgrade test.

**Comment by nprokopic** [comment]
Here you can see the local test run for custom cluster-aws branch https://github.com/giantswarm/cluster-aws/pull/431#issuecomment-1832192622

**Review by fiunchinho — APPROVED** [review]

**Comment by AverageMarcus** [comment]
The tests should now be able to pass if we run them, yes?

/run cluster-test-suites

**Comment by AverageMarcus** [comment]
CAPV MC is currently having unrelated issues so those failing tests are not due to this change. Adding skip label.

**Review by AverageMarcus — APPROVED** [review]
@nprokopic Could you please coordinate merging / releasing this with the cluster-aws changes?

**Comment by nprokopic** [comment]
> @nprokopic Could you please coordinate merging / releasing this with the cluster-aws changes?

cluster-aws release changes have already been merged, this is the fix we need next.

The next steps are then to cut a new release of cluster-aws and then fix the upgrade test for CAPA.
